### ISSUE TITLE
无法复现结果，急需帮助

### ISSUE THREAD

**Issue by kangkang189** [YuHengsss/YOLOV#37]
请问下可以告诉我们完整的训练命令吗，如何可以训练出87.5的结果呢，我torch版本和cuda已经保持了一致，但是无法复现结果，如果不是多节点训练的命令应该怎么实现呢？

**Comment by YuHengsss** [comment]
Hello, you should add strong augmentation while training to get 85+Ap50.  To get 87.5AP50, you should also take post-processing into account!

**Comment by kangkang189** [comment]
but i just run the shell "python tools/vid_train.py -f exps/yolov/yolov_x.py -c weights/yolov_x.pth -b 16 -d 2",does something wrong with it ? i close the multi node training ,my environment is "1card8gpus",what the shell do i use ? thanks!

**Comment by YuHengsss** [comment]
> but i just run the shell "python tools/vid_train.py -f exps/yolov/yolov_x.py -c weights/yolov_x.pth -b 16 -d 2",does something wrong with it ? i close the multi node training ,my environment is "1card8gpus",what the shell do i use ? thanks!

You could train it with single card. The epoch we set for yolov_x is 20 and you should get 85+AP@50 when you apply a formal testing. The eval process in training is only use part of validation set, please follow our instruction to do formal evaluation.

**Comment by kangkang189** [comment]
i just got 40+ap50 i think something wrong with it

**Comment by YuHengsss** [comment]
> i just got 40+ap50 i think something wrong with it

Do you use yolox pretrain weights?

**Comment by kangkang189** [comment]
-c weights/yolov_x.pth is it not the pretrain weights? i download from the readme

**Comment by YuHengsss** [comment]
> -c weights/yolov_x.pth is it not the pretrain weights? i download from the readme

Bro, you should read it carefully. What you use is the well-trained yolov weights not yolox pre-trained weights.

**Comment by kangkang189** [comment]
can you share the command you run with your work? i get use gpus=2 from your paper and 8imgs per

**Comment by YuHengsss** [comment]
> -c weights/yolov_x.pth is it not the pretrain weights? i download from the readme

And we suggest you to try small model first to verify you operation is correct

**Comment by kangkang189** [comment]
OH! thanks! but how can i get the yolox pre-trained weights

**Comment by kangkang189** [comment]
bro can we + wechat ,i just want to know how to get the 85+map and use it into my job!

**Comment by YuHengsss** [comment]
> OH! thanks! but how can i get the yolox pre-trained weights

We have already listed it in the readme file.

**Comment by YuHengsss** [comment]
> bro can we + wechat ,i just want to know how to get the 85+map and use it into my job!

You can try to contact the author of https://github.com/YuHengsss/YOLOV/issues/22#issue-1437400165. It seems he have finish the training precess successfully.

**Comment by kangkang189** [comment]
> > OH! thanks! but how can i get the yolox pre-trained weights
> 
> We have already listed it in the readme file.

i just see the well-trained weights in readme,where is the pretrained weights?

**Comment by YuHengsss** [comment]
> > > OH! thanks! but how can i get the yolox pre-trained weights
> > 
> > 
> > We have already listed it in the readme file.
> 
> i just see the well-trained weights in readme,where is the pretrained weights?

In the weights col of the main result table!!!!

**Comment by kangkang189** [comment]
but i 

> well-trained yolov weights

i just see the well-trained weights under main result table? i cant see the pretrained weights under main result table

**Comment by YuHengsss** [comment]
> but i
> 
> > well-trained yolov weights
> 
> i just see the well-trained weights under main result table? i cant see the pretrained weights under main result table

The yolox weights is the pretrained weights for yolov e.g: https://drive.google.com/file/d/1n8wkByqpHdrGy6z9fsoZpBtTa0I3JOcG/view?usp=sharing

**Comment by kangkang189** [comment]
!!! thanks! you are my hero!!!

**Comment by kangkang189** [comment]
hello,the results in paper mationed is AP50 85.5,AP50 means IOU0.50~0.95 or IOU0.50? I just run the yolov_s model under your code found IOU0.50~0.95 AP is 0.49 and the IOU0.50 AP is 0.74,which one means the "AP50" in your paper?

**Comment by kangkang189** [comment]
hello,the results in paper mationed is AP50 85.5,AP50 means IOU0.50-0.95 or IOU0.50? I just run the yolov_s model under your code found IOU0.50-0.95 AP is 0.49 and the IOU0.50 AP is 0.74,which one means the "AP50" in your paper?

**Comment by YuHengsss** [comment]
> hello,the results in paper mationed is AP50 85.5,AP50 means IOU0.50-0.95 or IOU0.50? I just run the yolov_s model under your code found IOU0.50-0.95 AP is 0.49 and the IOU0.50 AP is 0.74,which one means the "AP50" in your paper?

AP50 means AP@50. You can use the best_ckpt weights and set the gframe to 32 when testing which could boost the AP.

**Comment by kangkang189** [comment]
Thanks ! I will try it again!

**Comment by kangkang189** [comment]
i got the 76.9 AP50 with yolovs,how about the yolovx training hyper? training with 2Gpus or 1GpuS and 16batch/Gpus?

**Comment by YuHengsss** [comment]
> i got the 76.9 AP50 with yolovs,how about the yolovx training hyper? training with 2Gpus or 1GpuS and 16batch/Gpus?

You can try to modify the code and train it with 2 GPU. We train YOLOV-x just by one GPU.

**Comment by kangkang189** [comment]
hello,i want to know how to train yolo with 2 GPUs,i use "CUDA_VISIBLE_DEVICES=4,5 python tools/vid_train.py -f exps/yolov/yolov_x.py -c weights_pretrain/yoloxx_vid.pth --fp16 -b 16 -d 2"but i found the memory used in every gpu is equal,it means it dosent distribute the code?Which shell should i run?

**Comment by YuHengsss** [comment]
Please refer to https://github.com/YuHengsss/YOLOV/issues/35#issuecomment-1363664251
